ir/memory.cpp

```diff
@@ -307,7 +323,24 @@ void Memory::memcpy(const expr &d, const expr &s, const expr &bytes,
   Pointer dst(*this, d), src(*this, s);
   dst.is_dereferenceable(bytes, align_dst);
   src.is_dereferenceable(bytes, align_src);
-  // TODO
+  src.is_disjoint(bytes, dst, bytes);
+
+  uint64_t n;
+  if (bytes.isUInt(n) && n <= 4) {
+    for (unsigned i = 0; i < n; ++i) {
+      auto src_i = (src + i).release();
+      auto dst_i = (dst + i).release();
+      blocks_val = blocks_val.store(dst_i, blocks_val.load(src_i));
+    }
+  } else {
+    string name = "#idx_" + to_string(last_idx_ptr++);
+    Pointer idx(*this, expr::mkVar(name.c_str(), dst.bits()));
+
+    expr cond = idx.uge(dst).both() && idx.ult(dst + bytes).both();
+    expr val = expr::mkIf(cond, blocks_val.load((src + idx.get_offset()).release()),
```

Comment:
this lines seems over 80 characters

Reply:
not sure the condition for the src load is correct. It assumes that offset of src = 0.
Remember that memcpy both src and dst may start in the middle of an object.
So we need src + "delta between start of copy and idx". Which may not be `idx.get_offset()` if the memcpy wasn't done the start of the object.
What do you think?

Reply:
Thank you for your comment. I fixed this bug.